src/main/python/run_trecqa.py

```diff
@@ -5,219 +5,294 @@
+    questions, answers, labels_actual = load_data(args.input)
+    threshold = 0.7
+    # TODO: what is this ^^ magic number
+
+    qrel_file = ""
+
+    if args.qrel:
+        qrel_file = args.qrel
+    elif args.eval == 'jaccard':
+        qrel_file = '{}/qrel.jaccard.txt'.format(args.output)
+        create_qrel_jaccard(qrel_file, answers)
+    elif args.eval == 'pattern':
+        qrel_file = '{}/qrel.pattern.txt'.format(args.output)
+
+        stopwords = get_stopwords()
+
+        with open(qrel_file, 'w') as qrel:
+            for qid, question in zip(answers.keys(), questions):
+                # retrieve all sentences corresponding to h0 hits
+                all_sentences = pyserini.get_all_sentences(question, args.h0)
+                cleaned_sentences = clean_sentences(question, all_sentences, stopwords)
+                try:
+                    create_qrel_pattern(cleaned_sentences, pattern_dict[qid], qrel, qid)
+                except KeyError as e:
+                    print(e)
+                    print("pattern not found")
+
+    output_file = '{}/run.qa.{}.{}.h0_{}.h1_{}'.format(args.output, args.model,  args.eval, args.h0, args.h1)
+    seen_docs = []
+
+    with open(output_file, 'w') as out:
+        for qid, question in zip(answers.keys(), questions):
+            # sentence re-ranking after ad-hoc retrieval
+            candidates = get_answers(pyserini, question, int(args.h0), int(args.h1), args.model, args.index, args.w2v_cache,
+                                   args.qa_model_file)
+
+            if args.eval == 'jaccard':
+                scored_candidates = score_candidates(candidates, answers[qid])
+                i, unjudged_count = 0, 0
+
+                for key, value in scored_candidates.items():
+                  jaccard_similarity = value[1]
+                  i += 1
+
+                # check if the answer already exists in the TrecQA test set
+                if jaccard_similarity >= threshold:
+                    doc_id = value[0][0]
+                    if doc_id not in seen_docs:
+                        out.write('{} Q0 {} {} {} TRECQA\n'.format(qid, doc_id, i, value[2]))
+                        seen_docs.append(doc_id)
+                    else:
+                        unjudged_count += 1
+                        doc_id = 'unjudged{}'.format(unjudged_count)
+                        out.write('{} Q0 {} {} {} TRECQA\n'.format(qid, doc_id, i, value[2]))
+
+            elif args.eval == 'pattern':
+                if not args.pattern:
+                    print("Path to the pattern file should be included if the method of evaluation is pattern based")
+                    args.print_help()
+                    exit()
+                try:
+                    eval_by_pattern(qid, candidates, pattern_dict[qid], out)
+                except KeyError as e:
+                    print("Pattern not found for question: {}".format(e))
+
+        eval_depth = [int(i) for i in args.k.strip().split()]
```

Comment:
Could be replaced by `map(int, args.k.strip().split())` ... also I'm not sure how argparse deals with space separated arguments? I assume that this works as intended.